modules/linear_boltzmann_solvers/executors/pi_keigen.cc

```diff
@@ -182,5 +191,78 @@ PowerIterationKEigen::SetLBSScatterSource(const std::vector<double>& input,
     front_gs_, q_moments_local_, input, lbs_solver_.DensitiesLocal(), source_flags);
 }
 
+void
+PowerIterationKEigen::WriteRestartData(bool force)
+{
+  // Check if restarts are enabled
+  auto write_restart_time_interval = lbs_solver_.Options().write_restart_time_interval;
+  if (write_restart_time_interval == 0)
+    return;
+
+  // Check if enough time has elapsed since the last restart write
+  auto& last_restart_write_time = lbs_solver_.GetLastRestartWriteTime();
+  auto now = std::chrono::system_clock::now().time_since_epoch();
+  size_t now_secs = std::chrono::duration_cast<std::chrono::seconds>(now).count();
+  if (((now_secs - last_restart_write_time) < write_restart_time_interval) and not force)
+    return;
+
+  std::string fbase = lbs_solver_.Options().write_restart_directory_name + "/" +
+                      lbs_solver_.Options().write_restart_file_stem;
+  std::string fname = fbase + std::to_string(opensn::mpi_comm.rank()) + ".r";
```

Comment:
One thing I like to do when using hdf5, is the use of double extension, so here you could do `.r.h5`, or `.restart.h5`.
`.r` seems to be reserved for [R programming language](https://en.wikipedia.org/wiki/R_(programming_language)).

Reply:
I would lean towards `.restart.h5`.

Reply:
Fixed.

Reply:
Same as Zach

Reply:
Agreed. I have a new version coming up shortly. All the comments have been great, and I like the new version much better.